### PR TITLE
DOC: Adjust whatsnew

### PR DESCRIPTION
**Pull request by phofl**
Cleaned up some whatsnews and put the down casting change into a separate section, that's potentially noisy so it should be easy to find

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
nice - I agree with splitting this out. thanks!

File: doc/source/whatsnew/v2.2.0.rst [hunk 1]
```diff
@@ -386,6 +386,33 @@ For example:
...
+
+- :meth:`Series.replace` and :meth:`DataFrame.replace`
+- :meth:`DataFrame.fillna`, :meth:`Series.fillna`
+- :meth:`DataFrame.ffill`, :meth:`Series.ffill`
+- :meth:`DataFrame.bfill`, :meth:`Series.bfill`
+
+Explicitly call :meth:`DataFrame.infer_objects` to replicate the current behavior in the future.
+
+```
+result = result.infer_objects(copy=False)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:407]:
```suggestion
.. code-block:: ipython

    In [9]: result = result.infer_objects(copy=False)

File: doc/source/whatsnew/v2.2.0.rst [hunk 2]
```diff
@@ -386,6 +386,33 @@ For example:
...
+```
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:408]:
```suggestion
```

File: doc/source/whatsnew/v2.2.0.rst [hunk 3]
```diff
@@ -386,6 +386,33 @@ For example:
...
+
+Set the following option to opt into the future behavior:
+
+```
+pd.set_option("future.no_silent_downcasting", True)
+```
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:414]:
```suggestion
.. code-block:: ipython

    In [9]: pd.set_option("future.no_silent_downcasting", True)
```

File: doc/source/whatsnew/v2.2.0.rst (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v2.2.0.rst:407]: I would do without the `In [9]` here, given this is not mimicking a runnable snippet, just a generic line of example code

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**